app/Http/Controllers/TagController.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Tag;
+use App\Http\Requests\TagRequest;
+
+class TagController extends Controller
+{
+
+    
+    public function store(TagRequest $request)
+    {
+        $tag = Tag::create([
+            'name' => $request->name,
+        ]);
+
+        return response()->json($tag, 201);
+    }
+
+    public function update(TagRequest $request, $id)
+    {
+        $tag = Tag::findOrFail($id);
+
+        $tag->update([
+            'name' => $request->name,
+        ]);
+
+        return response()->json($tag, 200);
+    }
```

Comment:
```suggestion
    public function update(TagRequest $request, Tag $tag)
    {
        $tag->update($request->valdiated());

        return response()->json($tag, 200);
    }
```
Create Laravel resource for `$tag` - https://laravel.com/docs/12.x/eloquent-resources